src/Microsoft.Data/ArrowStringDataFrameColumn.cs

```diff
@@ -469,5 +469,55 @@ protected internal override Delegate GetDataViewGetter(DataViewRowCursor cursor)
 
         private ValueGetter<ReadOnlyMemory<char>> CreateValueGetterDelegate(DataViewRowCursor cursor) =>
             (ref ReadOnlyMemory<char> value) => value = this[cursor.Position].AsMemory();
+
+        /// <summary>
+        /// Returns a boolean column that is the result of an elementwise equality comparison of each value in the column with <paramref name="value"/>
+        /// </summary>
+        public PrimitiveDataFrameColumn<bool> ElementwiseEquals(string value)
+        {
+            ReadOnlySpan<byte> bytes = value != null ? Encoding.UTF8.GetBytes(value) : default(ReadOnlySpan<byte>);
+            PrimitiveDataFrameColumn<bool> ret = new PrimitiveDataFrameColumn<bool>(Name, Length);
+            for (long i = 0; i < Length; i++)
+            {
+                var strBytes = GetBytes(i);
+                ret[i] = value == null ? !IsValid(i) : strBytes.SequenceEqual(bytes);
```

Comment:
I would re-write this with 2 separate loops. There is no need to call `GetBytes(i)` when value is null. And there is no need to keep re-checking if value is null inside of the loop.

```C#
if (value == null)
{
    for (long i = 0; i < Length; i++)
    {
        ret[i] = !IsValid(i);
    }
}
else
{
    ReadOnlySpan<byte> bytes = Encoding.UTF8.GetBytes(value);
    for (long i = 0; i < Length; i++)
    {
        var strBytes = GetBytes(i);
        ret[i] = strBytes.SequenceEqual(bytes);
    }
}
```